core/src/main/java/ar/com/wolox/wolmo/core/service/WoloxRetrofitServices.java

```diff
@@ -15,43 +15,46 @@
 public abstract class WoloxRetrofitServices {
 
     private Retrofit mRetrofit;
-    private Retrofit mApiaryRetrofit;
     private Map<Class, Object> mServices;
 
     public void init() {
         mServices = new HashMap<>();
-        Gson gson = GsonBuilder.getBasicGsonBuilder().create();
-
-        HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
-        loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
-
-        OkHttpClient client = new OkHttpClient.Builder()
-                .addInterceptor(new SecuredRequestInterceptor())
-                .addInterceptor(loggerInterceptor)
-                .build();
-
         mRetrofit = new Retrofit.Builder()
                 .baseUrl(getApiEndpoint())
-                .addConverterFactory(GsonConverterFactory.create(gson))
-                .client(client)
-                .build();
-        mApiaryRetrofit = new Retrofit.Builder()
-                .baseUrl(getApiaryEndpoint())
-                .addConverterFactory(GsonConverterFactory.create(gson))
-                .client(client)
+                .addConverterFactory(GsonConverterFactory.create(getGson()))
+                .client(getOkHttpClient())
                 .build();
     }
 
+    /**
+     * Returns the API endpoint.
+     *
+     * @return URL endpoint
+     */
     public abstract String getApiEndpoint();
 
-    public abstract String getApiaryEndpoint();
+    /**
+     * Returns an instance of Gson to use for conversion.
+     *
+     * @return new Gson instance
+     */
+    protected Gson getGson() {
+        return GsonBuilder.getBasicGsonBuilder().create();
+    }
 
-    public <T> T getApiaryService(Class<T> clazz) {
-        T service = (T) mServices.get(clazz);
-        if (service != null) return service;
-        service = mApiaryRetrofit.create(clazz);
-        mServices.put(clazz, service);
-        return service;
+    /**
+     * Build and configure a new instance of OkHttpClient.
+     * You must add interceptors and configure the client before returning it.
+     *
+     * @return A configured instance of OkHttpClient
+     */
+    protected OkHttpClient getOkHttpClient() {
+        HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
+        loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+
+        return new OkHttpClient.Builder()
+                .addInterceptor(loggerInterceptor)
```

Comment:
How would you customize the builder on subclasses?
I was thinking of a structure like this:
```java
protected OkHttpClient getOkHttpClient() {
    OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
    initClient(okHttpClientBuilder);
}

protected initClient(OkHttpClient.Builder okHttpClientBuilder) {
    HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
    loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
    okHttpClientBuilder.addInterceptor(loggerInterceptor);
}
```
and subclasses would just do:
```java
@Override
protected initClient(OkHttpClient.Builder okHttpClientBuilder) {
    super.initClient(okHttpClientBuilder); // If they want the default client configuration
    // Custom code here
    ...
}
```
Maybe we can do the same for the `GsonBuilder`, and document how `initClient` and `initGson` can be overriden to customize it